File: benchmarks/lua/alloc/object_churn.lua
-- Object churn benchmark
-- Create N small tables with 3 fields and sum all values

local N = tonumber(arg and arg[1]) or 10000000
-- collectgarbage("collect")

local sum = 0
for i = 0, N - 1 do
    local obj = { x = i, y = i * 2, z = i * 3 }
    sum = sum + obj.x + obj.y + obj.z
end

print(string.format("%d", sum))

File: benchmarks/js/alloc/object_churn.js
// Object churn benchmark
// Create N small objects with 3 fields and sum all values

var print = typeof console !== 'undefined' ? console.log.bind(console) : globalThis.print;
var DEFAULT = 1000000;
var N = parseInt(typeof scriptArgs !== 'undefined' ? scriptArgs[1] : (typeof process !== 'undefined' ? process.argv[2] : '0')) || DEFAULT;

var sum = 0;
for (var i = 0; i < N; i++) {
    var obj = { x: i, y: i * 2, z: i * 3 };
    sum += obj.x + obj.y + obj.z;
}

print(sum);
